ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/assertions/CommandDeduplicationAssertions.scala

```diff
@@ -51,37 +55,93 @@ object CommandDeduplicationAssertions {
+          )
         } else {
-          fail("This case is not yet supported by this assertion")
+          assertReportedOffsetForAcceptedSubmission(
+            reportedHexOffset,
+            requestedDuration,
+            completionResponse,
+            ledger,
+            reportedOffsetCompletionStreamRequest,
+          )
         }
       case DurationSupport.Unrecognized(_) =>
         fail("Unrecognized deduplication duration support")
     }
   }
 
+  private def assertReportedOffsetForDuplicateSubmission(
+      reportedOffset: Ref.HexString,
+      completionResponse: CompletionResponse,
+      ledger: ParticipantTestContext,
+      reportedOffsetCompletionStreamRequest: CompletionStreamRequest,
+  )(implicit executionContext: ExecutionContext) = {
+    WithTimeout(5.seconds)(
+      ledger
+        .findCompletion(reportedOffsetCompletionStreamRequest)(c =>
+          c.commandId == completionResponse.completion.commandId && c.getStatus.code == Code.OK.value
+        )
+    ).map { optAcceptedCompletionResponse =>
+      val acceptedCompletionResponse = assertDefined(
+        optAcceptedCompletionResponse,
+        s"No accepted completion with the command ID '${completionResponse.completion.commandId}' since the reported offset $reportedOffset has been found",
+      )
+      assert(
+        acceptedCompletionResponse.offset.getAbsolute < completionResponse.offset.getAbsolute,
+        s"No accepted completion with the command ID '${completionResponse.completion.commandId}' before the completion's offset ${completionResponse.offset} has been found",
+      )
+    }
+  }
+
+  private def assertReportedOffsetForAcceptedSubmission(
+      reportedOffset: Ref.HexString,
+      requestedDuration: Duration,
+      completionResponse: CompletionResponse,
+      ledger: ParticipantTestContext,
+      reportedOffsetCompletionStreamRequest: CompletionStreamRequest,
+  )(implicit executionContext: ExecutionContext) = {
+    WithTimeout(5.seconds)(
+      ledger
+        .findCompletion(reportedOffsetCompletionStreamRequest)(
+          _.commandId == completionResponse.completion.commandId
+        )
+    ).map { optReportedOffsetCompletionResponse =>
+      val reportedOffsetCompletionResponse = assertDefined(
+        optReportedOffsetCompletionResponse,
+        s"No completion with the command ID '${completionResponse.completion.commandId}' since the reported offset $reportedOffset has been found",
+      )
+      assert(
+        reportedOffsetCompletionResponse.offset == LedgerOffset.of(
+          LedgerOffset.Value.Absolute(reportedOffset)
+        ),
+        s"No completion with the reported offset $reportedOffset has been found, the ${reportedOffsetCompletionResponse.offset} offset has been found instead",
+      )
+      val durationBetweenReportedDeduplicationOffsetAndCompletionRecordTimes = Duration
+        .between(
+          reportedOffsetCompletionResponse.recordTime,
```

Comment:
As discussed on Slack, I'm using `recordTime`s for simplicity. So far, it was able to pass reliably on Canton on my machine. Also, initially, I was passing a map from offsets to `recordTime`s. @nicu-da pointed out that I can actually query the completion service with offsets in an inclusive manner by using `HexOffset.previous`, so I modified the assertions accordingly.

Reply:
Fair enough. Let's leave it this way for now if it works.

Reply:
@andreaslochbihler-da, the build should be green now. I know you said to merge this PR before yours (https://github.com/digital-asset/daml/pull/12488) but it will result in a merge conflict. Just to double-check, shall I merge this one first?

Reply:
Sure, please go ahead. I can take care of the merge conflict.